### PR TITLE
Fixes the height reverting issue #272

### LINKED ISSUE #272: Changes to layout height gets reverted after switching to another circuits
Changes to layout height gets reverted after switching to another circuits.

To Reproduce:
1. Create a circuit
2. Click "Edit layout", add a bunch of height, and save the change
3. Click "Edit layout" again, and confirm that the change is saved
4. Switch to another circuit, then switch back again
5. Click "Edit layout", and notice that the change of height is reverted. However, it shouldn't.

This bug only affects the height of a layout.

I'm not working on this issue.

### PR DESCRIPTION
**Pull request by NavonilDas**
Fixes #272

### Demo
converted

### CODE REVIEW

**Comment by satu0king** [comment]
I checked the code, actually, it happens because we call `globalScope.fixLayout();` in `switchCircuit()` in data.js. This was actually intended as the user might create multiple inputs/outputs and then delete them but clearly, this restricts the flexibility for the user. 

@fpintos what do you advise, should we remove this feature?

**Comment by fpintos** [comment]
> @fpintos what do you advise, should we remove this feature?

Yes, we should. I was hit more than once on by this one. :-) 
The layout should always be the size the user specifies it.

That being said, we should not lose the auto-grow ability when new pins are added.
In other words, perhaps the function that changes the layout size should grow, but never shrink the layout.

File: public/js/subcircuit.js [hunk 1]
```diff
@@ -190,6 +197,11 @@ SubCircuit.prototype.reset = function() {
         return;
     }
 
+    if(allLayouts[subcircuitScope.id]){
+        // Set the width and height
+        subcircuitScope.layout = Object.assign(allLayouts[subcircuitScope.id]);
+    }
+
     subcircuitScope.layout.height = subcircuitScope.layout.height;
```
Comment (fpintos) [inline public/js/subcircuit.js]: These two lines seem like a no-op. I wonder if there was a typo in the original code.
Maybe we don't need the allLayouts copy if from here we could get a hold to the updated subcircuit template.
I was looking for a pointer to it but I'm not familiar with this part yet, @satu0king , do you know which variable would be holding all the loaded circuits here?

Comment (fpintos) [reply public/js/subcircuit.js]: Ah, I did some investigation:

By the end of saveLayout() the globalScope.layout is correct (and it is the same as scopeList[<id>].layout). As we switch circuits, by the time SubCircuit.reset is called the scopeList[<id>].layout.height was already reset.

Looking for where it happens I found switchCircuit() calling Scope.prototype.fixLayout(); this is the one resetting the height.

I think we can fix the issue by just removing Scope.prototype.fixLayout (in logix.js) altogether and its call from switchCircuit() (in data.js).

File: public/js/layout.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Layout.js - all subcircuit layout related code is here
 
 // Function to toggle between layoutMode and normal Mode
+var allLayouts = {}; // Stores all layouts of SubCircuit
```
Comment (fpintos) [inline public/js/layout.js]: It looks like we don't need this to fix this bug.
Just remove Scope.prototype.fixLayout() and its call from switchCircuit().

Comment (satu0king) [reply public/js/layout.js]: Yes just removing the function call to `globalScope.fixLayout();` will fix this.

Comment (NavonilDas) [reply public/js/layout.js]: It would be better to remove just one line instead of adding more lines.

Thanks, @satu0king and @fpintos

**Pull request merged**